dothething/dothething/ClipsView.swift

```diff
@@ -35,71 +34,101 @@ struct ClipsView: View {
+                Text("rally")
+                    .font(.custom("Montserrat-Medium", size: 17))
+                    .padding(.bottom)
+                    .tracking(4)
+
+                if !clipsViewModel.errorText.isEmpty && clipsViewModel.clips.isEmpty {
+                    Text(clipsViewModel.errorText)
+                        .padding(.top)
+                        .foregroundColor(Color(UIColor.systemGray))
+                } else if clipsViewModel.isLoading {
+                    ProgressView()
+                        .progressViewStyle(.circular)
+                        .padding(.top)
                 }
-                .opacity(clipsViewModel.uploadDisabled ? 0.5 : 1)
-
-                // share button
-                Button(action: {
-                    clipsViewModel.shareButtonPressed()
-                }) {
-                    Text("Share")
+                
+                ScrollView {
+                    LazyVGrid(columns: [GridItem](repeating: GridItem(.flexible(), spacing: 4), count: 3), spacing: 4) {
+                        ForEach(clipsViewModel.clips, id: \.self) { clip in
+                            ClipView(clip: clip)
+                        }
+                        .frame(height: (192 / 108) * geometry.size.width / 3)
+                    }
+                    .padding(.leading, 4)
+                    .padding(.trailing, 4)
                 }
-                .padding(.trailing)
-                .frame(maxWidth: .infinity, alignment: .trailing)
-                .opacity(clipsViewModel.shareDisabled ? 0.5 : 1)
-            }
-
-            if !clipsViewModel.errorText.isEmpty && clipsViewModel.clips.isEmpty {
-                Text(clipsViewModel.errorText)
-                    .padding(.top)
-                    .foregroundColor(Color(UIColor.systemGray))
-            } else if clipsViewModel.isLoading {
-                ProgressView()
-                    .progressViewStyle(.circular)
-                    .padding(.top)
-            }
-            
-            ScrollView {
-                LazyVGrid(columns: [GridItem(.adaptive(minimum: 108, maximum: 108), spacing: 16)]) {
-                    ForEach(clipsViewModel.clips, id: \.self) { clip in
-                        ClipView(clip: clip)
+                .padding(.top, -12)
+                .padding(.bottom, -12)
+                
+                ZStack {
+                     Rectangle()
+                        .frame(height: 48)
+                        .foregroundColor(.accentColor)
+
+                    Text("code: \(clipsViewModel.code)")
+                        .font(.custom("Montserrat-Light", size: 24))
+                        .foregroundColor(.accentColor)
+                        .colorInvert()
+                        .onTapGesture {
+                            print("Code tapped")
+                            UIPasteboard.general.string = clipsViewModel.code
+                        }
+                }
```

Comment:
These changes introduce a new section at the bottom of the `ClipsView` that includes a button with dynamic text ("place a domino" or "share this rally") and the ability to perform the corresponding action.

This is a good addition, as it provides a clear call-to-action for the user and makes the primary actions easily accessible. The styling and layout of the button are also well-designed.

The use of the `clipsViewModel.buttonText` property to determine the button text is a nice touch, as it allows for easy customization and maintenance of the button's behavior.